include/alpaka/block/shared/st/BlockSharedMemStMember.hpp

```diff
@@ -54,12 +54,12 @@ namespace alpaka
                         void alloc() const
                         {
                             m_allocdBytes = allocPitch<T>();
+#if (defined ALPAKA_DEBUG_OFFLOAD_ASSUME_HOST) && (! defined NDEBUG)
+                            ALPAKA_ASSERT(m_allocdBytes + sizeof(T) <= m_capacity);
+#endif
                             uint8_t* buf = &m_mem[m_allocdBytes];
```

Comment:
While we are discussing this code: Which version would you consider is more readable, the current
```c++
&m_mem[m_allocdBytes]
```
or
```c++
m_mem + m_allocdBytes
```
?